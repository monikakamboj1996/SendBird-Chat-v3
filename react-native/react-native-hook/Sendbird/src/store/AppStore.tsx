import AsyncStorage from '@react-native-async-storage/async-storage';
import { createStore, applyMiddleware } from 'redux';

import { persistStore, persistReducer } from 'redux-persist';
import thunk from 'redux-thunk';
import RootReducer from '../reducer/RootReducer';

// Middleware: Redux Persist Config
const persistConfig = {
  // Root
  key: 'root',
  // Storage Method (React Native)
  storage: AsyncStorage,
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, RootReducer);

// Redux: Store
const store = createStore(persistedReducer, applyMiddleware(thunk));

// Middleware: Redux Persist Persister
let persistor = persistStore(store);
// Exports
export { store, persistor };
