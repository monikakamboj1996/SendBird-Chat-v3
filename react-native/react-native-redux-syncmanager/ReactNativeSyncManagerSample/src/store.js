import {createStore, compose, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducer';

export const store = createStore(reducers, {}, compose(applyMiddleware(thunk)));
