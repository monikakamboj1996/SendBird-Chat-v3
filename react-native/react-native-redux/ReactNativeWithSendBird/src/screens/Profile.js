import React, { Component } from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import { initProfile, getCurrentUserInfo, updateProfile } from '../actions';
import { Button, Avatar, FormLabel, FormInput, FormValidationMessage, Spinner } from '../components';

class Profile extends Component {
  static navigationOptions = ({ navigation }) => {
    const { params } = navigation.state;
    return {
      title: 'Profile',
      headerLeft: (
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingLeft: 14 }}
          icon={{
            name: 'chevron-left',
            type: 'font-awesome',
            color: '#7d62d9',
            size: 18
          }}
          backgroundColor="transparent"
          onPress={() => navigation.goBack()}
        />
      ),
      headerRight: (
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingRight: 14 }}
          color={'#7d62d9'}
          title="save"
          backgroundColor="transparent"
          onPress={() => {
            params.handleSave();
          }}
        />
      )
    };
  };

  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      profileUrl: '',
      nickname: ''
    };
  }

  componentDidMount() {
    this.props.navigation.setParams({ handleSave: this._onSaveButtonPress });
    this.props.initProfile();
    this.setState({ isLoading: true }, () => {
      this.props.getCurrentUserInfo();
    });
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    const { userInfo, isSaved } = this.props;

    if (userInfo && userInfo !== prevProps.userInfo) {
      const { profileUrl, nickname } = userInfo;
      const isLoading = false;
      this.setState({ profileUrl, nickname, isLoading });
    }
    if (isSaved && isSaved !== prevProps.isSaved) {
      this.props.navigation.goBack();
    }
  }

  _onNicknameChanged = nickname => {
    this.setState({ nickname });
  };

  _onSaveButtonPress = () => {
    this.props.updateProfile(this.state.nickname);
  };

  render() {
    return (
      <View style={styles.containerStyle}>
        <Spinner visible={this.state.isLoading} />
        <View
          style={{
            justifyContent: 'center',
            flexDirection: 'row',
            marginTop: 50,
            marginBottom: 50
          }}
        >
          <Avatar
            large
            rounded
            source={this.state.profileUrl ? { uri: this.state.profileUrl } : require('../img/icon_sb_68.png')}
          />
        </View>

        <FormLabel labelStyle={[styles.defaultMargin, { marginTop: 20, fontSize: 13, fontWeight: '400' }]}>
          Nickname
        </FormLabel>
        <FormInput
          containerStyle={styles.defaultMargin}
          selectionColor={'#000'}
          inputStyle={{ color: '#000' }}
          value={this.state.nickname}
          maxLength={12}
          onChangeText={this._onNicknameChanged}
        />
        <FormValidationMessage labelStyle={{ marginLeft: 14 }}>{this.props.error}</FormValidationMessage>
      </View>
    );
  }
}

const styles = {
  containerStyle: {
    backgroundColor: '#fff',
    flex: 1
  },
  defaultMargin: {
    marginLeft: 14,
    marginRight: 14
  }
};

function mapStateToProps({ profile }) {
  const { userInfo, error, isSaved } = profile;
  return { userInfo, error, isSaved };
}

export default connect(
  mapStateToProps,
  { initProfile, getCurrentUserInfo, updateProfile }
)(Profile);
