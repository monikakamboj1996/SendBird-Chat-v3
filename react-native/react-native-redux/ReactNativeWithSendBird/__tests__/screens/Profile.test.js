/* @jest-environment jsdom */
import 'react-native';
import React from 'react';
import Profile from '../../src/screens/Profile';
import store from '../../src/store';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import 'jest-enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

let navParams = {
  channelUrl: 'channelUrl',
  isOpenChannel: true
};

const navigation = {
  setParams: params => {
    Object.assign(navParams, params);
  },
  navigate: jest.fn(),
  state: navParams,
  dispatch: jest.fn(),
  addListener: jest.fn()
};

describe('screen/Profile', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Profile store={store} navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
