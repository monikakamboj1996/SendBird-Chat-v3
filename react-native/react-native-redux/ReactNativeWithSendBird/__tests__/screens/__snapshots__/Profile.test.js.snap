// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen/Profile renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 50,
        "marginTop": 50,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 75,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 75,
          },
          Object {
            "borderRadius": 37.5,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "rgba(0,0,0,0.2)",
              "bottom": 0,
              "flex": 1,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "borderRadius": 37.5,
            },
            undefined,
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "test_profileUrl",
            }
          }
          style={
            Array [
              Object {
                "height": 75,
                "width": 75,
              },
              Object {
                "borderRadius": 37.5,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            Array [
              Object {
                "marginLeft": 14,
                "marginRight": 14,
              },
              Object {
                "fontSize": 13,
                "fontWeight": "400",
                "marginTop": 20,
              },
            ],
            undefined,
          ],
        ]
      }
    >
      Nickname
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 14,
        "marginRight": 14,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      maxLength={12}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      selectionColor="#000"
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "color": "#000",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="test_nickname"
    />
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#ff190c",
              "fontSize": 15,
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 5,
            },
            Object {
              "marginLeft": 14,
            },
            undefined,
          ],
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
